src/ACEtk/interpretation/ContinuousEnergyNeutron/Builder/EnergyDistribution/LevelScattering/Builder.hpp

```diff
@@ -0,0 +1,21 @@
+template< typename ParentBuilder >
+class Builder: public Base::Builder< Builder< ParentBuilder >, ParentBuilder > {
+  using BaseBuilder = Base::Builder< Builder, ParentBuilder >;
+
+  friend class Base::Builder< Builder, ParentBuilder >;
+
+  std::optional< double > atomicWeightRatio_;
+  std::optional< double > QValue_;
+
+protected:
+  LevelScattering construct(){
+    return { QValue_.value(), atomicWeightRatio_.value() };
```

Comment:
Some as the DiscretePhotonEnergy::construct(): you can get into uncaught exceptions (contrary to other construct() functions).

Reply:
Fixed.